pymeasure/instruments/velleman/velleman_k8090.py

```diff
@@ -0,0 +1,249 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import serial
+
+from pymeasure.instruments import Instrument
+
+
+def _parse_channels(channels) -> str:
+    """Convert array of channel numbers into mask if needed."""
+    if isinstance(channels, list):
+        mask = 0x00
+        for ch in channels:
+            mask |= 1 << (ch - 1)
+    else:
+        mask = channels
+
+    return hex(mask)
+
+
+def _get_process_status(items):
+    """Process the result of a 0x51 status message."""
+    if len(items) < 4 or items[0] != 0x51:
+        return None
+
+    return _ints_to_bool_lists(items[1:])
+
+
+def _ints_to_bool_lists(numbers):
+    """Convert 8-bit integers into lists of booleans.
+
+    Also works for a single given integer.
+
+    The least significant bit (#0) will be first
+    """
+    if isinstance(numbers, int):
+        numbers = [numbers]
+
+    bool_lists = [[bool(num & (1 << n)) for n in range(8)] for num in numbers]
+
+    if len(bool_lists) == 1:
+        return bool_lists[0]
+
+    return bool_lists
```

Comment:
Maybe an [IntFlag enum](https://docs.python.org/3/library/enum.html#intflag) could be useful to set/read which relays are set.

Reply:
You mean in the API signature or internally? I.e. a user would type something like:

```
from pymeasure.instruments.velleman import VellemanK8090, Channels

instrument.switch_on(Channels.CH1 | Channels.CH2 | Channels.CH3)
instrument.switch_off(Channels.ALL)
```

It's a bit longer, but I do think it's more robust. I'll change it.

Reply:
Exactly.